lib/src/pubspec.dart

```diff
@@ -293,8 +298,18 @@ class Pubspec extends PubspecBase {
           pubspecPath);
     }
 
+    String? overridesContents;
```

Comment:
Actually information about the overrides probably doesn't belong in the pubspec at all.

Only a root node can have pubspec_overrides, so we should probably have it in the entrypoint along with pubspec and lockfile.

The version solver should then take an explicit map of overrides that can be given from Entrypoint.acquireDependencies

Reply:
I added a `withPubspecOverrides` option to `Package.load` and `Pubspec.load`, which is only set to `true` by `Entrypoint(String rootDir, this.cache)`.

All the code to parse dependency specifications is in `Pubspec`, which we would need to parse `pubspec_overrides.yaml`. Factoring that code out to parse `pubspec_overrides.yaml` separately seems a bit overkill to me, but we can do that, too.

Reply:
I have reworked the PR to parse `pubspec_overrides.yaml` into a separate `PubspecOverrides` file, which gets rid of the hacky `OverrideYamlMap`.

Seems to me that `Package` is a more natural fit to store the loaded `PubspecOverrides`. Quite a few places use `Package.dependencyOverrides` and this where we can apply the `dependencyOverrides` from `PubspecOverrides`.
I had to update the `PackageLister` slightly, so that is uses `Package.dependencyOverrides`.

WDYT?